### DISCUSSION TITLE
Interpolation weights for arbitrary physical points

### DISCUSSION THREAD

**Discussion by elliotenglish** [libMesh/libmesh#3745] [Q&A] [answered]
I am implementating an explicit semi-lagrangian advection scheme (in addition to using an implicit scheme for other terms) using libmesh for the underlying data representation. I've been reading through the code in order to determine how to compute the shape functions for arbitrary physical points, but it looks like the shape() method takes local coordinates only. Given this, I looked for a method to go from physical coordinates to local coordinates, but can't find explicitly anything to do this. I'm probably missing something obvious, but could someone point me in the right direction to figure this out? Thanks!

**Comment by jwpeterson** [comment, marked as answer]
> Given this, I looked for a method to go from physical coordinates to local coordinates, but can't find explicitly anything to do this.

See `FEMap::inverse_map()`

**Reply by elliotenglish** [reply]
Awesome, thank you for the quick turn around on that. I entirely missed looking at the static function dox.
